tempesta_fw/sock.c

```diff
@@ -1009,6 +1011,10 @@ ss_connect(struct sock *sk, struct sockaddr *uaddr, int uaddr_len, int flags)
 	if (sk->sk_state != TCP_CLOSE)
 		return -EISCONN;
 
+	/* Prevent possible deadlock with inet_get_local_port_range() (#578) */
+	if (spin_is_locked(&net->ipv4.ip_local_ports.lock.lock))
+		return -EAGAIN;
```

Comment:
Looks very hacky: locking retries are done on timeout basis requireing many code to execute for each lock try. Also timeout retries are adjusted just to do retries, while typically timers are for periodic works.